sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -361,9 +361,12 @@ object StateStore extends Logging {
     val storeProvider = loadedProviders.synchronized {
       startMaintenanceIfNeeded()
       val provider = loadedProviders.getOrElseUpdate(
-        storeProviderId,
-        StateStoreProvider.createAndInit(
-          storeProviderId.storeId, keySchema, valueSchema, indexOrdinal, storeConf, hadoopConf)
+        storeProviderId, {
```

Comment:
Nit: Not sure why not add this before `provider` init? Maybe it would be less cryptic but no strong opinion.

Reply:
Yeah that sounds better. I think we can place in prior to create provider to avoid creating provider if schema validation fails.